File: jbake-core/src/main/java/org/jbake/template/model/PublishedPagesExtractor.java
package org.jbake.template.model;

import org.jbake.app.ContentStore;
import org.jbake.app.DocumentList;
import org.jbake.model.DocumentModel;
import org.jbake.template.ModelExtractor;

public class PublishedPagesExtractor implements ModelExtractor<DocumentList<DocumentModel>> {

    @Override
    public DocumentList<DocumentModel> get(ContentStore db, TemplateModel model, String key) {
        return db.getPublishedPages();
    }

}

File: jbake-core/src/main/java/org/jbake/template/model/PublishedPagesExtractor.kt
package org.jbake.template.model

import org.jbake.app.DocumentList
import org.jbake.template.TypedModelExtractor

class PublishedPagesExtractor : TypedModelExtractor<DocumentList<*>> {

    override fun extract(context: RenderContext, key: String): DocumentList<*> {
        return context.publishedPages
    }
}
